core/plugins/org.polarsys.capella.core.data.core.properties/src/org/polarsys/capella/core/data/core/properties/fields/TitleBlockBasicElementGroup.java

```diff
@@ -0,0 +1,204 @@
+  }
+
+  /**
+   * @param parent
+   * @param widgetFactory
+   * @param hasNameField
+   * @param hasContentField
+   */
+  public TitleBlockBasicElementGroup(Composite parent, TabbedPropertySheetWidgetFactory widgetFactory,
+      boolean hasNameField, boolean hasContentField) {
+    super(widgetFactory);
+
+    Group textGroup = widgetFactory.createGroup(parent, ICommonConstants.EMPTY_STRING);
+    textGroup.setLayout(new GridLayout(2, false));
+    GridData gd = new GridData(GridData.FILL_HORIZONTAL);
+    gd.horizontalSpan = 2;
+    textGroup.setLayoutData(gd);
+
+    // Name
+    if (hasNameField) {
+      nameTextField = createTextField(textGroup, Messages.getString("NamedElement.NameLabel"));
+    }
+    // Content
+    if (hasContentField) {
+      contentTextField = createTextField(textGroup, Messages.getString("NamedElement.ContentLabel"));
+    }
+  }
+
+  /**
+   * @param textGroup
+   * @param textLabel
+   */
+  private Text createTextField(Group textGroup, String textLabel) {
+    widgetFactory.createCLabel(textGroup, textLabel);
+
+    Text textField = widgetFactory.createText(textGroup, ICommonConstants.EMPTY_STRING);
+    textField.addFocusListener(this);
+    textField.addKeyListener(this);
+    textField.addModifyListener(this);
+    textField.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+
+    return textField;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public void loadData(EObject semanticElement, String name, String content) {
+    super.loadData(semanticElement, null);
+
+    if (null != semanticElement) {
+      if (null != nameTextField)
+        setTextValue(nameTextField, name);
+      if (null != contentTextField)
+        setTextValue(contentTextField, content);
+    }
+  }
+
+  protected void setTextValue(Text text, String value) {
+    text.setText(value);
+  }
+
+  /**
+   * @param textField
+   *          text field to be filled
+   */
+  @Override
+  protected void fillTextField(Text textField) {
+    if (textField.equals(nameTextField)) {
+      setNameValue(semanticElement, nameTextField.getText());
+    } else if (textField.equals(contentTextField)) {
+      setContentValue(semanticElement, contentTextField.getText());
+    }
+  }
+
+  private void setNameValue(EObject object, final Object value) {
+    if (!((DAnnotation) object).getDetails().get("Name:").equals(value.toString())) {
+      DDiagram diagram = (DDiagram) object.eContainer();
+      AbstractReadWriteCommand command = new AbstractReadWriteCommand() {
+        @Override
+        public void run() {
+          ((DAnnotation) object).getDetails().put("Name:", value.toString());
+          RefreshRepresentationsCommand refreshCommand = new RefreshRepresentationsCommand(
+              TransactionUtil.getEditingDomain(diagram), new NullProgressMonitor(), diagram);
+          refreshCommand.execute();
+        }
+      };
+      executeCommand(command);
+    }
+  }
+
+  private void setContentValue(EObject object, final Object value) {

```

Comment:
This method is called for KeyPressed events which makes the expression evaluated for each key press. This can cause performance issue and not really necessary. Focus lost event should be good.

Reply:
I added this because focus listener seems to have some problems. This problem is also present for example for PropertyValue. If you complete the Name and then complete Summary and then click somewhere on the screen, then if you go back to PropertyValue, the Summary is not saved.